frontend/workflows/projectSelector/src/helpers.tsx

```diff
@@ -96,7 +103,75 @@ const exclusiveProjectDependencies = (
   return { upstreams, downstreams };
 };
 
+// deriveHiddenStatus evaluates whether to hide an upstream/downstream based on
+// the checked status of/exclusivity to projects in Group.Projects
+const deriveHiddenStatus = (state: State, group: Group, key: string): boolean => {
+  // if true, don't need to go through the flow below as we only hide an upstream/downstream
+  if (group === Group.PROJECTS) {
+    return false;
+  }
+
+  const unchecked = [];
+  _.forIn(state[Group.PROJECTS], (v, k) => {
+    if (!v.checked) {
+      // collect the unchecked projects, if any
+      unchecked.push(k);
+    }
+  });
+
+  // no unchecked projects
+  if (unchecked.length === 0) {
+    updateHidden(state, group, key, false);
```

Comment:
this is buggy/doesn't always accurately reflect current state of hidden projects. still thinking about how to properly update the global state

Reply:
actually the bug was with how I was handling some edge cases w.r.t hiding an upstream/downstream if all its parent projects are unchecked. will push up the fixes